src/process/cli/pcap.rs

```diff
@@ -0,0 +1,324 @@
+                    &InterfaceDescriptionBlock {
+                        linktype: DataLink::ETHERNET,
+                        snaplen: 0xffff,
+                        options: vec![
+                            InterfaceDescriptionOption::IfName(Cow::Owned(format!(
+                                "{} ({})",
+                                ifname, netns
+                            ))),
+                            InterfaceDescriptionOption::IfDescription(Cow::Owned(match ifindex {
+                                0 => "Fake interface".to_string(),
+                                _ => format!("ifindex={}", ifindex),
+                            })),
+                        ],
+                    }
+                    .into_block(),
+                )?;
+
+                let id = self.ifaces.len() as u32;
+                self.ifaces.insert(key, id);
+                id
+            }
+        };
+
+        // Add the packet itself.
+        self.writer.write_block(
+            &EnhancedPacketBlock {
+                interface_id: id,
+                timestamp: Duration::from_nanos(common.timestamp),
+                original_len: packet.len,
+                data: Cow::Borrowed(&packet.packet),
+                options: vec![EnhancedPacketOption::Comment(Cow::Owned(format!(
+                    "probe={}:{}",
+                    &kernel.probe_type, &kernel.symbol
+                )))],
+            }
+            .into_block(),
+        )?;
+
+        Ok(())
+    }
+
+    /// Report parser statistics. Should be called after processing was
+    /// completed.
+    fn report_stats(&self) {
+        info!("{} event(s) were processed", self.stats.processed);
+
+        if self.stats.missing_skb != 0 {
+            warn!(
+                "{} event(s) were skipped because of missing skb information",
+                self.stats.missing_skb
+            );
+        }
+        if self.stats.missing_packet != 0 {
+            warn!(
+                "{} event(s) were skipped because of missing raw packet",
+                self.stats.missing_packet
+            );
+        }
+        if self.stats.missing_dev != 0 {
+            warn!(
+                "{} event(s) are using a fake net device (no device information was found)",
+                self.stats.missing_dev
+            );
+        }
+        if self.stats.missing_ns != 0 {
+            warn!(
+                "{} event(s) are using a fake netns (no netns information was found)",
+                self.stats.missing_ns
+            );
+        }
+    }
+}
+
+/// Generate a PCAP file from a collection of recorded events.
+#[derive(Parser, Debug, Default)]
+#[command(name = "pcap")]
+pub(crate) struct Pcap {
+    #[arg(
+        short,
+        long,
+        help = "Filter events from this probe. Probes should follow the TYPE:TARGET pattern. See `retis collect --help` for more details on the probe format."
+    )]
+    pub(super) probe: String,
```

Comment:
apparently, wireshark supports filtering for this:
`frame.comment == "probe=raw_tracepoint:net:netif_receive_skb"`

Not explicitly asking for it, but it might be convenient in some cases to have all packets from different probes (although they could appear as duplicated) and then filter from the GUI. It would spare some back and forth.